pkg/runner/client_pool.go

```diff
@@ -0,0 +1,56 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type pool struct {
+	sync.Mutex
+
+	availableC chan *kubernetes.Clientset
```

Comment:
Not draining this channel on close is fine for now, because we have no concept of graceful shutdown across components, but at one point we'll want to do that.